certbot/error_handler.py

```diff
@@ -19,6 +19,14 @@
              signal.SIGXCPU, signal.SIGXFSZ])
 
 
+class SignalExit(Exception):
+    """This exception is used stop of execution of the code in the body of the
+    ErrorHandler context manager.
+
+    """
+    pass
+
+
 class ErrorHandler(object):
     """Registers functions to be called if an exception or signal occurs.
```

Comment:
With the change of raising an exception in the signal handler, this class can really only be used as a context manager. I think this is the right approach, however, we should update the docstring below to say that and make all methods besides `register` "protected" by adding a leading underscore.